### PR TITLE
check if fields are valid during parse.

### LINKED ISSUE #3525: Inequalities in the SELECT clause should return a parse error
Using any of `=`,`!=`,`<`,`>`,`<=`,`>=` will yield empty results for all types.

### PR DESCRIPTION
**Pull request by DanielMorsing**
Binary expressions that yield a boolean are invalid and we can catch
these at parse time.

Fixes #3525

### CODE REVIEW

File: influxql/parser.go [hunk 1]
```diff
@@ -1476,6 +1476,11 @@ func (p *Parser) parseField() (*Field, error) {
 	if err != nil {
 		return nil, err
 	}
```
Comment (dgnorton) [inline influxql/parser.go]: Mark the position here...

Comment (DanielMorsing) [inline influxql/parser.go]: Thought about it, but the invalid expression might be deep into the field. Example:

`select value * (2 > 3) * 43 from foo`

The position information will be lost by the time that the parser yields the expression at the end of the 43 token.

Comment (dgnorton) [inline influxql/parser.go]: But at least they would know which field expression it was in if they had something like this...

Comment (dgnorton) [inline influxql/parser.go]: Another idea: the `validField` visitor could store the operator and the error message could read something like `invalid operator < found in field expression at <position>`

**Comment by dgnorton** [comment]
lgtm

**Comment by dgnorton** [comment]
Oh...need to update `CHANGELOG.md`

File: influxql/parser_test.go [hunk 2]
```diff
@@ -1244,6 +1244,9 @@ func TestParser_ParseStatement(t *testing.T) {
 		{s: `select derivative() from myseries`, err: `invalid number of arguments for derivative, expected at least 1 but no more than 2, got 0`},
 		{s: `select derivative(mean(value), 1h, 3) from myseries`, err: `invalid number of arguments for derivative, expected at least 1 but no more than 2, got 3`},
 		{s: `SELECT field1 from myseries WHERE host =~ 'asd' LIMIT 1`, err: `found asd, expected regex at line 1, char 42`},
+		{s: `SELECT value > 2 FROM cpu`, err: `invalid operator for field > at line 1, char 8`},
```
Comment (pauldix) [inline influxql/parser_test.go]: would be better to have the actual name in the error. Also mention that it's an invalid operator for the select clause. Maybe something like:

```
invalid operator in select clause: > at line 1, char 8. these belong in the where clause
```

**Comment by pauldix** [comment]
+1

**Pull request merged**